### PR TITLE
ddl: add test case of local temporary table

### LINKED ISSUE #25943: Forbid setting tiflash in local temporary tables

### PR DESCRIPTION
**Pull request by sylzd**
### What problem does this PR solve?

Issue Number: close #25943 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by djshow832 — APPROVED** [review]

File: ddl/db_test.go [hunk 1]
```diff
@@ -5725,16 +5725,16 @@ func (s *testSerialDBSuite) TestSetTiFlashReplicaForTemporaryTable(c *C) {
...
 	tk.MustGetErrCode("alter table temp set tiflash replica 1", errno.ErrOptOnTemporaryTable)
-	tk.MustExec("drop table temp")
+	tk.MustGetErrCode("alter table temp2 set tiflash replica 1", errno.ErrUnsupportedDDLOperation)
+	tk.MustExec("drop table temp, temp2")
 
-	tk.MustExec("drop table if exists normal")
 	tk.MustExec("create table normal(id int)")
 	defer tk.MustExec("drop table normal")
-	tk.MustExec("alter table normal set tiflash replica 1")
-	tk.MustQuery("select REPLICA_COUNT from information_schema.tiflash_replica where table_schema='test' and table_name='normal'").Check(testkit.Rows("1"))
```
Comment (lcwangchao) [inline ddl/db_test.go:5737]: It's a case for `create table like` should not inherit normal table's tiflash replica option when the created table is a temporary table.  I don't think you should remove it and you should add a same test for local temporary table.

You can try to exec `make failpoint-enable` before test. Without it, the case will fail

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**